include/uapi/linux/bpf.h

```diff
@@ -184,6 +184,7 @@ enum bpf_prog_type {
 	BPF_PROG_TYPE_STRUCT_OPS,
 	BPF_PROG_TYPE_EXT,
 	BPF_PROG_TYPE_LSM,
+	BPF_PROG_TYPE_IO_FILTER,
```

Comment:
Maybe all these things also need to be protected under #ifdef?

Reply:
Hm I am not sure, are you talking about an #ifdef on a Kconfig option?

Reply:
IIUC this is the userspace header so it should be here unconditionally -- if the kernel doesn't have io_filter enabled then it will just refuse to attach anything

Reply:
ah that's right.